ucm2/codecs/rt5640/DigitalMics.conf

```diff
@@ -10,7 +10,15 @@ SectionDevice."Mic" {
 		cset "name='Mono ADC MIXR ADC2 Switch' on"
 		cset "name='Stereo ADC MIXL ADC2 Switch' on"
 		cset "name='Stereo ADC MIXR ADC2 Switch' on"
-		cset "name='Internal Mic Switch' on"
+		If.Controls {
```

Comment:
At first of all, thank you for your work. It seems like a right track to make things more maintainable. The BayTrail (Intel) specific controls should be moved to the upper level configs. @plbossart , FYI,

Also, `If` blocks cannot be in the sequences. You can define [] arrays multiple times and the sequences will be added to the end of array. So this is fine:

````
EnableSequence [
   sequence3
]
If.x {
   True.EnableSequence [
      sequence1
   ]
}
If.y {
  False.EnableSequence [
     sequence2
  ]
}
````

Note that Include / If blocks are evaluated as first (see the sequence numbers). That's the reason why we have some dummy (empty) string `If` blocks in the configs.

Reply:
I pushed the updated version that implements yours suggestion, it works using the latest alsa-lib master. Please take a look.